apps/src/lib/node/ledger/ethereum_node/test_tools.rs

```diff
@@ -87,7 +87,7 @@ pub mod event_endpoint {
                         }
                     };
                     tracing::debug!("Serialized event - {:#?}", event);
-                    match sender.send(event) {
+                    match sender.try_send(event) {
```

Comment:
```suggestion
                    match tokio::block_on(sender.send(event)) {
```
not sure if this type checks, but something along these lines, to avoid losing events arbitrarily

Reply:
there is no such thing as `tokio:;block_on`. Do you mean `tokio_test::block_on`?

Reply:
https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#method.current

fetch the current handle to the runtime, then call `block_on`

Reply:
so instead of calling `map`, we would call `then`. this way we can still call async stuff from this ctx

Reply:
another nice option, idk if you prefer, is to use [`flume`](https://docs.rs/flume) channels, which have both a [sync](https://docs.rs/flume/latest/flume/struct.Sender.html#method.send) and [async](https://docs.rs/flume/latest/flume/struct.Sender.html#method.send_async) sendind method

Reply:
Took a little bit of finagling, but I got the `then` function to work.